addons/mail/models/mail_partner.py

```diff
@@ -0,0 +1,132 @@
+        create_context['default_user_id'] = False
+        if custom_values is None:
+            custom_values = {}
+        defaults = {
+            'name': msg.get('subject') or _("No Subject"),
+            'email_from': msg.get('from'),
+            'email_cc': msg.get('cc'),
+            'partner_id': msg.get('author_id', False),
+        }
+        defaults.update(custom_values)
+        thread = super(MailPartnerMixin, self.with_context(create_context)).message_new(msg, custom_values=defaults)
+        if(thread.message_subscription_strategy(msg) == self.ON_CREATE_STRATEGY): #if we want to subscribe emails on create
+            thread.partner_auto_subscribe(msg)
+        return thread
+
+    def message_update(self, msg, update_vals=None):
+        """ Overrides mail_thread update that is called by the mailgateway
+            through process.
+            This override create the partner partner_id if it doesnt exist based on the email_from
+        """
+        if(self.message_subscription_strategy(msg) == self.ON_REPLY_STRATEGY): #if we want to subscribe emails on update
+            self.partner_auto_subscribe(msg)
+        super(MailPartnerMixin,self).message_update(msg, update_vals)
+
+    def message_subscription_strategy(self,msg):
+        return self.ON_REPLY_STRATEGY
+
+    def emails_to_subscribe(self,msg):
+        """
+            Email to subscribe. A partner is create if it doesn't exists
+            Overwrite this method to add more email to subscribe. 
+            email_from should be present in the list to allow partner auto update with emails
+        """
+        return [self.email_from]
+    
+    def partner_emails_to_subscribe(self,msg):
+        """
+            Email to subscribe only if a partner exists
+            Overwrite this method to add more partner to subscribe. 
+        """
+        #TODO functionnality to confirm
+        return [] #tools.email_split(msg.get('cc')) # + [tools.email_split(self.email_cc)] ?
+
+    def partner_auto_subscribe(self,msg):
+        """
+            Subscribe some partners based on emails. 
+            -will subscribe and create partners if necessary for all emails in "emails_to_subscribe"
+            -will subscribe all email contained in "partner_emails_to_subscribe" if a corresponding partner exists
+        """
+        email_list = self.emails_to_subscribe(msg)
+        partner_ids = [p for p in self._find_partner_from_emails(email_list, force_create=True) if p]
+        self.message_subscribe(partner_ids)
+        #Case autosubscribe: a partner is created automaticaly when answering to an email
+        self.message_update_partner_id(self.env['res.partner'].browse(partner_ids))
+        partner_email_list = self.partner_emails_to_subscribe(msg)
+        real_partner_ids = [p for p in self._find_partner_from_emails(partner_email_list, force_create=False) if p]
+        self.message_subscribe(real_partner_ids)
+
+    def message_partner_id_field(self):
```

Comment:
Are there any model not using partner_id ? I think we could lessen a bit the call stack by directly checking partner_id and not giving possibility to override it in a method.

Reply:
I don't remember why I decided to put the field in a getter but it doesn't look to be necessary right now.